axon-server-connector/src/main/java/org/axonframework/axonserver/connector/ErrorCode.java

```diff
@@ -16,84 +16,99 @@
 
 package org.axonframework.axonserver.connector;
 
+import io.axoniq.axonserver.grpc.ErrorMessage;
+import org.axonframework.axonserver.connector.command.RemoteCommandException;
 import org.axonframework.axonserver.connector.event.util.EventStoreClientException;
+import org.axonframework.axonserver.connector.query.RemoteQueryException;
+import org.axonframework.axonserver.connector.util.ExceptionSerializer;
 import org.axonframework.commandhandling.CommandExecutionException;
+import org.axonframework.commandhandling.NoHandlerForCommandException;
+import org.axonframework.commandhandling.distributed.CommandDispatchException;
 import org.axonframework.common.AxonException;
 import org.axonframework.eventsourcing.eventstore.EventStoreException;
-import org.axonframework.modelling.command.AggregateRolledBackException;
+import org.axonframework.messaging.EventPublicationFailedException;
 import org.axonframework.modelling.command.ConcurrencyException;
+import org.axonframework.queryhandling.NoHandlerForQueryException;
 import org.axonframework.queryhandling.QueryExecutionException;
 
-import java.util.Arrays;
 import java.util.concurrent.TimeoutException;
+import java.util.function.BiFunction;
+
+import static java.util.Arrays.stream;
 
 
 /**
- * Converts an EventStoreClientException to the relevant Axon framework exception.
+ * Converts an Axon Server Error to the relevant Axon framework exception.
  */
 public enum ErrorCode {
     // Generic errors processing client request
-    AUTHENTICATION_TOKEN_MISSING("AXONIQ-1000", EventStoreException.class),
-    AUTHENTICATION_INVALID_TOKEN("AXONIQ-1001", EventStoreException.class),
-    NODE_IS_REPLICA("AXONIQ-1100", EventStoreException.class),
+    AUTHENTICATION_TOKEN_MISSING("AXONIQ-1000", AxonServerException::new),
+    AUTHENTICATION_INVALID_TOKEN("AXONIQ-1001", AxonServerException::new),
+
+    //Event publishing errors
+    INVALID_SEQUENCE("AXONIQ-2000", (code, error) -> new ConcurrencyException(error.getMessage(), new AxonServerException(code, error))),
```

Comment:
Can we mention EVENT somewhere in these enum instances? And COMMAND / QUERY in the others, so that it's clear from the API that these are related to C/E/Q

Reply:
Sure, I can change them into INVALID_EVENT_SEQUENCE, EVENT_PAYLOAD_TOO_LARGE and if you like NO_EVENT_STORE_MASTER_AVAILABLE (it is a bit too long but it contains the EVENT mention)